lib/3rd_party_adapters/LVGL/GuiEngine.cpp

```diff
@@ -0,0 +1,214 @@
+    // drawing commands to make them visible on screen!
+    display.display();
+
+    // Initialize lvgl library
+    static lv_disp_draw_buf_t draw_buf;
+    lv_init();
+    lv_disp_draw_buf_init(&draw_buf, buf.get(), NULL, configuration.screen_width * 16);
+
+    // Initialize the display driver for lvgl
+    static lv_disp_drv_t disp_drv;
+    lv_disp_drv_init(&disp_drv);
+    disp_drv.hor_res = configuration.screen_width;
+    disp_drv.ver_res = configuration.screen_height;
+    disp_drv.flush_cb = flushSSD1306Adafruit;
+    disp_drv.draw_buf = &draw_buf;
+    disp_drv.user_data = this;
+
+    // Register display driver to lvgl
+    lv_disp_drv_register(&disp_drv);
+    delay(200);
+
+    // Create monochomatic theme and set it as default
+    lv_theme_t *mono_theme = lv_theme_mono_init(0, false, &lv_font_unscii_8);
+    lv_disp_set_theme(0, mono_theme);
+
+    // create first text in lvgl
+    lv_obj_t *label = lv_label_create(lv_scr_act());
+    lv_label_set_text(label, "LVGL is up");
+    lv_obj_align(label, LV_ALIGN_TOP_MID, 0, 0);
+
+    // display lvgl screen
+    lv_timer_handler();
+}
+
+/**
+ * @brief implementation for lvgl read_cb for the left key
+ */
+static void keypad_read_left(lv_indev_drv_t *indev_drv, lv_indev_data_t *data)
+{
+    if (!myKeypad)
+        return;
+
+    // assign key state
+    data->state = (myKeypad->isKeyPressed(KeyId::LEFT)) ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
+    data->key = LV_KEY_PREV;
+}
+
+/**
+ * @brief implementation for lvgl read_cb for the right key
+ */
+static void keypad_read_right(lv_indev_drv_t *indev_drv, lv_indev_data_t *data)
+{
+    if (!myKeypad)
+        return;
+
+    // assign key state
+    data->state = (myKeypad->isKeyPressed(KeyId::RIGHT)) ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
+    data->key = LV_KEY_NEXT;
+}
+
+/**
+ * @brief implementation for lvgl read_cb for the enter key
+ */
+static void keypad_read_enter(lv_indev_drv_t *indev_drv, lv_indev_data_t *data)
+{
+    if (!myKeypad)
+        return;
+
+    // assign key state
+    data->state = (myKeypad->isKeyPressed(KeyId::ENTER)) ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
+    data->key = LV_KEY_ENTER;
+}
+
+/**
+ * @brief implementation for lvgl read_cb for the back key
+ */
+static void keypad_read_back(lv_indev_drv_t *indev_drv, lv_indev_data_t *data)
+{
+    if (!myKeypad)
+        return;
+
+    // assign key state
+    data->state = (myKeypad->isKeyPressed(KeyId::BACK)) ? LV_INDEV_STATE_PRESSED : LV_INDEV_STATE_RELEASED;
+    data->key = LV_KEY_ESC;
+}
+
+/**
+ * @brief register the Keypad to read the button states from
+ * 
+ * @param keypad - pointer to the Keypad object
+ */
+void GuiEngine::registerKeyPad(IKeypad *keypad)
+{
+    //assign keypad reference to local pointer
+    myKeypad = keypad;
+
+    // create default group for button navigation and assign input device to it
+    group = lv_group_create();
```

Comment:
```suggestion
    static lv_group_t *const group = lv_group_create();
```

Else it could lead to a memory leak.